src/system/SystemObject.cpp

```diff
@@ -66,28 +65,26 @@ DLL_EXPORT void Object::Release()
     }
 }
 
-DLL_EXPORT bool Object::TryCreate(Layer & aLayer, size_t aOctets)
+DLL_EXPORT bool Object::TryCreate(size_t aOctets)
 {
     bool lReturn = false;
 
-    if (__sync_bool_compare_and_swap(&this->mSystemLayer, nullptr, &aLayer))
+    if (__sync_bool_compare_and_swap(&this->mRefCount, 0, 1))
```

Comment:
nit: this lReturn usage makes the code harder to read. How about:

```cpp
if (!__sync_bool_compare_and_swap(&this->mRefCount, 0, 1)) {
  return false; // object already in use
}

this->AppState = nullptr;
memset(reinterpret_cast<char *>(this) + sizeof(*this), 0, aOctets - sizeof(*this));

return true;
```

Reply:
Acked